be/src/exprs/agg/combinator/agg_state_utils.h

```diff
@@ -28,17 +25,28 @@ class AggStateUtils {
 public:
     // scalar function: suffix for aggregate state combinator functions
     static constexpr const char* STATE_FUNCTION_SUFFIX = "_state";
+    static constexpr const char* STATE_UNION_FUNCTION_SUFFIX = "_state_union";
+    static constexpr const char* STATE_MERGE_FUNCTION_SUFFIX = "_state_merge";
 
     // aggregate function: suffixes for aggregate state combinator functions
     static constexpr const char* AGG_STATE_UNION_SUFFIX = "_union";
     static constexpr const char* AGG_STATE_MERGE_SUFFIX = "_merge";
+    static constexpr const char* AGG_STATE_COMBINE_SUFFIX = "_combine";
     static constexpr const char* AGG_STATE_IF_SUFFIX = "_if";
     static constexpr std::string FUNCTION_COUNT = "count";
 
     static bool is_agg_state_function(const std::string& func_name) noexcept {
         return !func_name.empty() && func_name.ends_with(STATE_FUNCTION_SUFFIX);
     }
 
+    static bool is_agg_state_union_function(const std::string& func_name) noexcept {
```

Comment:
is it do dispatch based on `enum` rather than the function name ? which can be more efficient and safe

Reply:
We needs to dispatch by `TFunction's TFunctionName`to determine this which TFunctionName is string inside, no need to use `enum` by now since it's a trival check.